src/_index.js

```diff
@@ -40,7 +39,20 @@ class Popup{
         this.options.hasMobileLayout = this.isBooleanOptionTrue(ATTRS.mobileLayout, this.options.hasMobileLayout);
 
         // get options from JSON init
-        this.options = getOptions(this, this.options);
+        this.id = options.id || this.el.id || DEFAULTS.id;
+
+        this.options = getOptionsFromAttribute(
+            {
+                target: this.el,
+                attributeName: ATTRS.init,
+                defaultOptions: {...this.options, id: this.id},
+                numericValues: ['autoShow'],
+                onIsString: (options) => {
```

Comment:
What's the point of this code?

Reply:
I have might made some mistakes. I have just pushed a new code replacing for these old codes. Is it good?